android/src/app/organicmaps/routing/RoutingBottomMenuController.java

```diff
@@ -265,9 +265,8 @@ private void showRouteAltitudeChart()
       mAltitudeChart.setImageBitmap(bm);
       UiUtils.show(mAltitudeChart);
       final String unit = limits.isMetricUnits ? mAltitudeDifference.getResources().getString(R.string.m) : mAltitudeDifference.getResources().getString(R.string.ft);
-      mAltitudeDifference.setText(String.format(Locale.getDefault(), "↗ %d %s ↘ %d %s",
-                                                limits.totalAscent, unit,
-                                                limits.totalDescent, unit));
+      mAltitudeDifference.setText("↗ " + limits.totalAscentString + " " + unit +
```

Comment:
Is + faster/better than String.format?

Reply:
Yes, the + operator seems to be much faster:

First column: String.format, second column: + operator
2778346 ns, 72961 ns
1656885 ns, 341115 ns
1513307 ns, 45154 ns
1233769 ns, 32039 ns

And as we're dealing now only with strings, the code seems to be clearer than with the String.format function.